packages/manager/apps/key-management-service/src/components/credential/create/CreateAddIdentities.component.tsx

```diff
@@ -0,0 +1,136 @@
+import { Subtitle } from '@ovhcloud/manager-components';
+import React, {
+  Dispatch,
+  SetStateAction,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import { useTranslation } from 'react-i18next';
+import { OsdsButton, OsdsIcon, OsdsText } from '@ovhcloud/ods-components/react';
+import {
+  ODS_BUTTON_SIZE,
+  ODS_BUTTON_VARIANT,
+  ODS_ICON_NAME,
+  ODS_ICON_SIZE,
+} from '@ovhcloud/ods-components';
+import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
+import { ShellContext } from '@ovh-ux/manager-react-shell-client';
+import { useNavigate, useParams } from 'react-router-dom';
+import { useIdentityData } from '@/hooks/credential/useIdentityData';
+import { ROUTES_URLS } from '@/routes/routes.constants';
+import IdentitiesRootAccount from './identities/IdentitiesRootAccount.component';
+import IdentitiesSelectedUsersList from './identities/IdentitiesSelectedUsersList.component';
+import IdentitiesSelectedGroups from './identities/IdentitiesSelectedGroups.component';
+import IdentitiesSelectedServiceAccounts from './identities/IdentitiesSelectedServiceAccounts.component';
+
+type CreateAddIdentitiesProps = {
+  identityURNs: string[];
+  setIdentityURNs: Dispatch<SetStateAction<string[]>>;
+  prevStep: () => void;
+  nextStep: () => void;
+};
+
+const CreateAddIdentities = ({
+  identityURNs,
+  setIdentityURNs,
+  prevStep,
+  nextStep,
+}: CreateAddIdentitiesProps) => {
+  const { t } = useTranslation('key-management-service/credential');
+  const navigate = useNavigate();
+  const { okmsId } = useParams();
+  const [isRootAccount, setIsRootAccount] = useState<boolean>(false);
+  const { userList, groupList, serviceAccountList } = useIdentityData();
+  const { nichandle } = useContext(ShellContext).environment.getUser();
+
+  useEffect(() => {
+    if (isRootAccount) {
+      setIdentityURNs([`urn:v1:eu:identity:account:${nichandle}`]);
+      return;
+    }
+    const identityURNsTmp = [];
+    const userURNs = userList.map((user) => user.urn);
+    const groupURNs = groupList.map((group) => group.urn);
+    const serviceAccountURNs = serviceAccountList.map(
+      (serviceAccount) => serviceAccount.identity,
+    );
+    identityURNsTmp.push(...userURNs, ...groupURNs, ...serviceAccountURNs);
+    setIdentityURNs(identityURNsTmp);
```

Comment:
```suggestion
    const userURNs = userList.map((user) => user.urn);
    const groupURNs = groupList.map((group) => group.urn);
    const serviceAccountURNs = serviceAccountList.map(
      (serviceAccount) => serviceAccount.identity,
    );
    setIdentityURNs([...userURNs, ...groupURNs, ...serviceAccountURNs]);
```
No need to create a temporal array I think